ui/src/dashboards/utils/time.ts

```diff
@@ -0,0 +1,30 @@
+interface InputTimeRange {
+  seconds?: number
+  lower?: string
+  upper?: string
+}
+
+interface OutputTimeRange {
+  since: number
+  until: number | null
+}
+
+export const millisecondTimeRange = ({
+  seconds,
+  lower,
+  upper,
+}: InputTimeRange): OutputTimeRange => {
+  // Is this a relative time range?
+  if (seconds) {
+    return {since: Date.now() - seconds * 1000, until: null}
+  }
+
+  const since = Date.parse(lower)
+  let until
+  if (upper === 'now()') {
+    until = Date.now()
+  } else {
+    until = Date.parse(upper)
+  }
```

Comment:
I know you don't like terniary's, but wouldn't 
``` 
const upper = (upper==='now()')
? Date.now()
: Date.parse(upper)
```
look soooo much better? ;)

completely aesthetic pref. obvs.